Installing clang_linux-64 and clangxx_linux-64 results in a non-working C++ compiler installed due to mismatching libstdxx packages installed

### Solution to issue cannot be found in the documentation.

- [x] I checked the documentation.

### Issue

Installing `clang_linux-64` and `clangxx_linux-64` packages results in a compiler that is not able to find C++ standard library headers due to the mismatch between the different version of gcc packages installed.

In particular, if you install `conda create -n test clang_linux-64 clangxx_linux-64` the following packages end up installed:

~~~
libgcc                      15.2.0           he0feb66_18           conda-forge
libgcc-devel_linux-64       15.2.0           hcc6f6b0_118          conda-forge
libgomp                     15.2.0           he0feb66_18           conda-forge
libstdcxx                   15.2.0           h934c35e_18           conda-forge
libstdcxx-devel_linux-64    13.4.0           h6963c3b_118          conda-forge
~~~

and during compiltion it results in error like:

~~~
 │ │ Configuration file: $BUILD_PREFIX/bin/x86_64-conda-linux-gnu.cfg
 │ │ Found candidate GCC installation: $BUILD_PREFIX/bin/../lib/gcc/x86_64-conda-linux-gnu/13.4.0
 │ │ Found candidate GCC installation: $BUILD_PREFIX/bin/../lib/gcc/x86_64-conda-linux-gnu/15.2.0
 │ │ Selected GCC installation: $BUILD_PREFIX/bin/../lib/gcc/x86_64-conda-linux-gnu/15.2.0
 │ │ Candidate multilib: .;@m64
 │ │ Selected multilib: .;@m64
 │ │  (in-process)
 │ │  "$BUILD_PREFIX/bin/clang-21" -cc1 -triple x86_64-conda-linux-gnu -E -disable-free -clear-ast-before-backend -disab
 │ │ le-llvm-verifier -discard-value-names -main-file-name null -mrelocation-model pic -pic-level 2 -pic-is-pie -mframe-
 │ │ pointer=all -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu x8
 │ │ 6-64 -tune-cpu generic -debugger-tuning=gdb -fdebug-compilation-dir=$SRC_DIR -v -fcoverage-compilation-dir=$SRC_DIR
 │ │  -resource-dir $BUILD_PREFIX/lib/clang/21 -isysroot $BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot -internal-isystem
 │ │ $BUILD_PREFIX/lib/clang/21/include -internal-isystem $BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot/usr/local/include
 │ │  -internal-isystem $BUILD_PREFIX/bin/../lib/gcc/x86_64-conda-linux-gnu/15.2.0/../../../../x86_64-conda-linux-gnu/in
 │ │ clude -internal-externc-isystem $BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot/include -internal-externc-isystem $BUI
 │ │ LD_PREFIX/x86_64-conda-linux-gnu/sysroot/usr/include -source-date-epoch 1771852053 -std=c++17 -fdeprecated-macro -f
 │ │ error-limit 19 -fgnuc-version=4.2.1 -fskip-odr-check-in-gmf -fcxx-exceptions -fexceptions -faddrsig -D__GCC_HAVE_DW
 │ │ ARF2_CFI_ASM=1 -o - -x c++ /dev/null
 │ │ clang -cc1 version 21.1.8 based upon LLVM 21.1.8 default target x86_64-conda-linux-gnu
 │ │ ignoring nonexistent directory "$BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot/usr/local/include"
 │ │ ignoring nonexistent directory "$BUILD_PREFIX/bin/../lib/gcc/x86_64-conda-linux-gnu/15.2.0/../../../../x86_64-conda
 │ │ -linux-gnu/include"
 │ │ ignoring nonexistent directory "$BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot/include"
 │ │ #include "..." search starts here:
 │ │ #include <...> search starts here:
 │ │  $BUILD_PREFIX/lib/clang/21/include
 │ │  $BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot/usr/include
 │ │ End of search list.
 │ │ # 1 "/dev/null"
 │ │ # 1 "<built-in>" 1
 │ │ # 1 "<built-in>" 3
 │ │ # 476 "<built-in>" 3
 │ │ # 1 "<command line>" 1
 │ │ # 1 "<built-in>" 2
 │ │ # 1 "/dev/null" 2
 │ │ main.cpp:1:10: fatal error: 'iostream' file not found
 │ │     1 | #include <iostream>
 │ │       |          ^~~~~~~~~~
~~~

that you can easily reproduce with  https://gist.github.com/traversaro/de0fd7c1b80072b06509164a2c22903b, i.e. :

~~~
git clone https://gist.github.com/traversaro/de0fd7c1b80072b06509164a2c22903b repr
cd repr
pixi run build-broke
# See how pinning libstdcxx-devel_linux-64 to 15.* fixes the issue
~~~

It seems that clang gets confused if there are different bits of different gcc installation.

Manually installing `libstdcxx-devel_linux-64 15.*` seems to fix the issue.

### Installed packages

```shell
(test) straversaro@gblaptop0007:~$ conda list
# packages in environment at /home/straversaro/.pixi/envs/conda/envs/test:
#
# Name                      Version          Build                 Channel
_openmp_mutex               4.5              20_gnu                conda-forge
binutils_impl_linux-64      2.45.1           default_hfdba357_101  conda-forge
binutils_linux-64           2.45.1           default_h4852527_101  conda-forge
clang-21                    21.1.8           default_h99862b1_3    conda-forge
clang_impl_linux-64         21.1.8           default_h666bb00_3    conda-forge
clang_linux-64              21.1             hba70d61_21           conda-forge
clangxx_impl_linux-64       21.1.8           default_h666bb00_3    conda-forge
clangxx_linux-64            21.1             h8edcda7_21           conda-forge
compiler-rt                 21.1.8           ha770c72_1            conda-forge
compiler-rt21               21.1.8           hb700be7_1            conda-forge
compiler-rt21_linux-64      21.1.8           hffcefe0_1            conda-forge
compiler-rt_linux-64        21.1.8           ha770c72_1            conda-forge
icu                         78.2             h33c6efd_0            conda-forge
kernel-headers_linux-64     6.12.0           he073ed8_5            conda-forge
ld_impl_linux-64            2.45.1           default_hbd61a6d_101  conda-forge
libclang-cpp21.1            21.1.8           default_h99862b1_3    conda-forge
libcompiler-rt              21.1.8           hb700be7_1            conda-forge
libgcc                      15.2.0           he0feb66_18           conda-forge
libgcc-devel_linux-64       15.2.0           hcc6f6b0_118          conda-forge
libgomp                     15.2.0           he0feb66_18           conda-forge
libiconv                    1.18             h3b78370_2            conda-forge
libllvm21                   21.1.8           hf7376ad_0            conda-forge
liblzma                     5.8.2            hb03c661_0            conda-forge
libstdcxx                   15.2.0           h934c35e_18           conda-forge
libstdcxx-devel_linux-64    13.4.0           h6963c3b_118          conda-forge
libxml2                     2.15.1           he237659_1            conda-forge
libxml2-16                  2.15.1           hca6bf5a_1            conda-forge
libzlib                     1.3.1            hb9d3cd8_2            conda-forge
sysroot_linux-64            2.39             hc4b9eeb_5            conda-forge
tzdata                      2025c            hc9c84f9_1            conda-forge
zstd                        1.5.7            hb78ec9c_6            conda-forge
```

### Environment info

```shell
(test) straversaro@gblaptop0007:~$ conda info

     active environment : test
    active env location : /home/straversaro/.pixi/envs/conda/envs/test
            shell level : 1
       user config file : /home/straversaro/.condarc
 populated config files : /home/straversaro/.condarc
          conda version : 25.11.1
    conda-build version : not installed
         python version : 3.13.11.final.0
                 solver : libmamba (default)
       virtual packages : __archspec=1=skylake
                          __conda=25.11.1=0
                          __cuda=12.8=0
                          __glibc=2.39=0
                          __linux=6.6.87.2=0
                          __unix=0=0
       base environment : /home/straversaro/.pixi/envs/conda  (writable)
      conda av data dir : /home/straversaro/.pixi/envs/conda/etc/conda
  conda av metadata url : None
           channel URLs :
          package cache : /home/straversaro/.pixi/envs/conda/pkgs
                          /home/straversaro/.conda/pkgs
       envs directories : /home/straversaro/.pixi/envs/conda/envs
                          /home/straversaro/.conda/envs
               platform : linux-64
             user-agent : conda/25.11.1 requests/2.32.5 CPython/3.13.11 Linux/6.6.87.2-microsoft-standard-WSL2 ubuntu/24.04.3 glibc/2.39 solver/libmamba conda-libmamba-solver/25.11.0 libmambapy/2.5.0
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

--- a/recipe/recipe.yaml
+++ b/recipe/recipe.yaml
@@ -19,6 +19,7 @@ context:
   # https://github.com/gcc-mirror/gcc/commit/9e18a25331fa25c3907249fede65a02c6817b06e
   last_symbol_bump: 12
   min_runtime_version: ${{ gcc_major if ((gcc_major | int) > (last_symbol_bump | int)) else last_symbol_bump }}
+  library_prefix: ${{ "Library/" if win else "" }}
 
 recipe:
   name: ctng-compiler-activation

--- a/recipe/recipe.yaml
+++ b/recipe/recipe.yaml
@@ -1,7 +1,7 @@
 schema_version: 1
 
 context:
-  build_num: 21
+  build_num: 22
   gcc_version: ${{ gcc_version | default("13.1.0") }}
   gcc_major: ${{ (gcc_version | split("."))[0] | int }}
   default_skip: ${{ (target_platform == "win-64" and cross_target_platform != "win-64") or (cross_target_platform == "win-64" and gcc_major|int < 13) or cross_target_platform == "linux-s390x" or s390x or ((cross_target_platform in ("osx-64", "osx-arm64") or osx) and gcc_major|int < 15) }}
@@ -222,6 +222,7 @@ outputs:
         - ${{ pin_subpackage("clang_impl_" ~ cross_target_platform, exact=True) }}
         # C++ compiler on linux needs a stdlib
         - libstdcxx-devel_${{ cross_target_platform }} ${{ gcc_version }}.*
+        - libgcc-devel_${{ cross_target_platform }} ${{ gcc_version }}.*
         - clangxx ${{ clang_version }}.*
     tests:
       - script:

--- a/recipe/recipe.yaml
+++ b/recipe/recipe.yaml
@@ -126,6 +126,8 @@ outputs:
         # for activation of binutils env vars
         - binutils_${{ cross_target_platform }}
         - clang_impl_${{ cross_target_platform }} ${{ clang_version }}.*
+        # need to ensure libgcc-devel_<target> matching the run-exports is used
+        - libgcc-devel_${{ cross_target_platform }} ${{ gcc_version }}.*
       run_exports:
         strong:
           - libgcc >=${{ min_runtime_version }}

